Issue#157 turnstile updates

russell-d-e opened the pull request:
Creating initial pull request. 

This included automated ingestion for everything in the 02-Software folder.  Additional node groups will be added to automate the ingestion of data from 01-System and 03-Verification.

Ingestion is fully automated with the exception of running the SADL queries to populate the CSV files

glguy commented on Turnstile-Ontology/99-Utils/CsvTemplates/Ingest-Threads.csv:
```diff
@@ -0,0 +1 @@
+identifier_SoftwareDesign, identifier_SYSTEM_0, identifier_SYSTEM, 
```
This is an example of a CSV file I'm not sure about exactly, but the trailing comma makes me wonder if something went wrong.

russell-d-e replied:
Nothing wrong this matches the Spec.  The trailing comma was is just an artifact of when I generate the CSV template the script is not smart about only adding a comma when needed, it just for every property adds the "{Sparql_ID},".  I have the script to only include the comma between column headers.

russell-d-e replied:
> * Need to change the comment for TOOL in AGENTS.sadl (we renamed the class).
> * Good to provide alias for DevelopmentPlan.sadl, CounterApplicationReviews.sadl, DevelopmentPlanData.sadl and maybe some other files.
> * Should we take this opportunity to align Turnstiles and Turnstile; note we have "Turnstiles is a SYSTEM, has identifier "Turnstile"."?

Updated TOOL notes to not be software specfic, added alias and harmonized them throughout the sadl files. updated Turnstile system definitions to be consistent in all sadl files for both planning and system design.

glguy commented on Turnstile-Ontology/02-Software/03-Implementation/.output.o.rack:
```diff
@@ -0,0 +1,12 @@
+:- multifile build_with/5, build_from/2, build_inputs/2, build_outputs/2,
```
These `.rack` files are generated data that shouldn't be committed.

russell-d-e replied:
removed .rack file

glguy commented on Turnstile-Ontology/02-Software/03-Implementation/.counter.d:
```diff
@@ -0,0 +1 @@
+counter.exe: counter.c counter.h input.h hw.h output.h
```
I believe these `.d` files shouldn't be committed.

russell-d-e replied:
removed .d files

glguy reviewed (changes requested):
`Turnstile-Ontology/99-Utils/README.md` is empty, is that intentional?

`input.o` and `output.o` and `counter.exe` are binary files that should not be committed.

russell-d-e commented:
> `Turnstile-Ontology/99-Utils/README.md` is empty, is that intentional?
> 
> `input.o` and `output.o` and `counter.exe` are binary files that should not be committed.

I missed the README.md it was updated with a description of the generation process and automation.  
Removed  .exe and .o

AbhaMoitra commented:
- Need to change the comment for TOOL in AGENTS.sadl (we renamed the class).

-  Good to provide alias for DevelopmentPlan.sadl, CounterApplicationReviews.sadl, DevelopmentPlanData.sadl and maybe some other files.

- Should we take this opportunity to align Turnstiles and Turnstile; note we have "Turnstiles is a SYSTEM, has identifier "Turnstile"."?

russell-d-e commented on RACK-Ontology/ontology/AGENTS.sadl:
```diff
@@ -35,6 +35,10 @@ employedBy (note "Relates people to the company they were acting on behalf of.")
 employedBy is a type of actedOnBehalfOf.
 
 
+TOOL
```
not sure if it is an important distinction, but I was thinking of a TOOL as a piece of software that performs a specific activity. for an example a complier or a build system.  the description for SOFTWAREAGENT seemed much more in scope broad. I did not include just as this change already has so much and I think it would need some more thought put to it but I would expect TOOL to add properties like version or TQL. The description given for SOFTWARE agent seemed to almost be describing an OS or a computer.

In general I think there is some improvements that could be made around agents.  My proposal is for now to just remove SOFTWAREAGENT and in the future we can work the improve/identify what types of AGENTS should be included.

russell-d-e replied:
I think this actually has be OBE after Abha's comment pointed out that a TOOL may be more than just software.  the notes for TOOL is being updated to:

`An aid (software, hardware, etc) that automates the performance of a specific activity.`

I still wonder if SOFTWAREAGENT should be removed but I did not originate it so I will just leave it be for now unless you want me to remove it.

glguy replied:
we can keep it for now

russell-d-e commented:
> * Need to change the comment for TOOL in AGENTS.sadl (we renamed the class).
> * Good to provide alias for DevelopmentPlan.sadl, CounterApplicationReviews.sadl, DevelopmentPlanData.sadl and maybe some other files.
> * Should we take this opportunity to align Turnstiles and Turnstile; note we have "Turnstiles is a SYSTEM, has identifier "Turnstile"."?


Updated TOOL notes to not be software specfic, added alias and harmonized them throughout the sadl files. updated Turnstile system definitions to be consistent in all sadl files for both planning and system design.
